### DISCUSSION TITLE
How to tele-operate the robot remotely using a joystick?

### DISCUSSION THREAD

**Discussion by eecue** [iRobotEducation/create3_docs#99] [Questions] [answered]
I'm running joy on a remote system: `ros2 launch teleop_twist_joy teleop-launch.py joy_config:='ps4' `

I can see the topic and detect my movements on the system connected to the create3 but it doesn't move:

```header:
  stamp:
    sec: 1656808475
    nanosec: 594846161
  frame_id: joy
axes:
- -0.0
- 0.8767923712730408
- -0.0
- -0.0
- 0.0
- 0.0
buttons:
```

I see the topic: `ros2 topic list`

```
ros2 topic list
/battery_state
/cmd_audio
/cmd_lightring
/cmd_vel
/dock
/hazard_detection
/imu
/interface_buttons
/ir_intensity
/ir_opcode
/joy
/joy/set_feedback
/kidnap_status
/mouse
/odom
/parameter_events
/robot_state/transition_event
/rosout
/slip_status
/static_transform/transition_event
/stop_status
/system_monitor/transition_event
/tf
/tf_static
/wheel_status
/wheel_ticks
/wheel_vels
```

**Comment by justinIRBT** [comment]
To isolate the joystick program from commanding velocity on the `cmd_vel` topic, can you try the following command line command:
```
ros2 topic pub -r 20 /cmd_vel geometry_msgs/msg/Twist "{linear: {x: 0.2, y: 0.0, z: 0.0}, angular: {x: 0.0, y: 0.0, z: 0.0}}"
```
to see if that drives the robot.  If that drives the robot, another thing to try is the create3_examples configuration in this example:
https://github.com/iRobotEducation/create3_examples/tree/galactic/create3_teleop
If it doesn't drive the robot, then there is likely an issue with your DDS config, sometimes you can see the topic list, but when you try to interact with topics, such as doing
```
ros2 topic echo /dock
```
it won't work.  In that case, see configuration options here:
https://iroboteducation.github.io/create3_docs/setup/xml-config/

**Reply by eecue** [reply]
Yep all of those commands work fine as well as the keyboard teleop.

**Reply by justinIRBT** [reply]
Did you try the create3_teleop version/configuration of the joystick node linked above.  I haven’t used it myself, so I am not sure if it configures something differently than the defaults in the joystick package.  Can you also confirm what the joystick node is publishing

**Reply by eecue** [reply]
Figured it out, I wasn't pressing the enable button.

**Comment by eecue** [comment, marked as answer]
Ok I figured it out, I wasn't pressing the enable button.
